zone/bot.cpp

```diff
@@ -5655,30 +6062,48 @@ bool Bot::DoFinishedSpellSingleTarget(uint16 spell_id, Mob* spellTarget, EQ::spe
 		}
 
 		if (!noGroupSpell) {
-			Group *g = GetGroup();
-			if (g) {
-				for (int i = 0; i < MAX_GROUP_MEMBERS; i++) {
-					if (g->members[i]) {
-						if ((g->members[i]->GetClass() == Class::Necromancer) && (IsEffectInSpell(thespell, SE_AbsorbMagicAtt) || IsEffectInSpell(thespell, SE_Rune))) {
-						}
-						else
-							SpellOnTarget(thespell, g->members[i]);
+			std::vector<Mob*> v;
+
+			if (RuleB(Bots, RaidBuffing)) {
+				v = GetSpellTargetList();
+			}
+			else {
+				v = GatherSpellTargets(false, spellTarget);
+			}
```

Comment:
This could be a helper function since this is reused a few places and then referenced inline with the loops

Example

```cpp
for (auto &m : GetBotSpellTargets(false, spellTarget))
```